### PR TITLE
isPrimaryKey on RealmObjectSchema

### LINKED ISSUE #2440: Add `isPrimaryKey` and `getPrimaryKey` to `RealmObjectSchema`
We used to have `Table.getPrimaryKey()` to return the PK row. But this cannot be achieved by the new Dynamic API anymore.

Propose to add two more methods to `RealmObjectSchema`

* `void RealmObjectSchema.isPrimaryKey(String fieldName)`
* `String RealmObjectSchema.getPrimaryKey()`

Requested from SO http://stackoverflow.com/questions/35978812/realm-io-v0-88-get-primary-key-column-name

### PR DESCRIPTION
**Pull request by beeender**
* Add RealmObjectSchema.isPrimaryKey(). RealmObjectSchema.getPrimaryKey()
      is not added since it doesn't seem to be necessary.
* isNullable & isRequired didn't throw which is wrong.
* Add throws doc to hasIndex.
    
Close #2440

### CODE REVIEW

**Comment by stk1m1** [comment]
:+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## 0.89.0
+
+### Enhancements
+
+* RealmObjectSchema.isPrimaryKey(String) (#2440)
+
+### Bug fixes
+
+* RealmObjectSchema.isRequired and RealmObjectSchema.isNullable don't throw when the give field names don't exist.
```
Comment (cmelchior) [inline CHANGELOG.md]: Use `()` after method names. It makes it a lot more readable IMO

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java [hunk 2]
```diff
@@ -16,24 +16,42 @@
...
     private RealmSchema realmSchema;
 
-    @Override
-    protected void setUp() throws Exception {
-        super.setUp();
-        RealmConfiguration realmConfig = new RealmConfiguration.Builder(getContext()).build();
+    @Before
+    public void setUp() {
+        RealmConfiguration realmConfig = configFactory.createConfiguration();
         Realm.deleteRealm(realmConfig);
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java]: Remove this

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java [hunk 3]
```diff
@@ -43,9 +61,8 @@ protected void setUp() throws Exception {
         schema = realmSchema.create("NewClass");
     }
 
-    @Override
-    protected void tearDown() throws Exception {
-        super.tearDown();
+    @After
+    public void tearDown() throws Exception {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java]: No longer throws Exception

**Comment by cmelchior** [comment]
Apart from a few minor comments :+1:

File: CHANGELOG.md (hunk 1 above)
Comment (MGaetan89) [inline CHANGELOG.md]: `when the given field name doesn't exist.`?

Comment (jcj52436999) [inline CHANGELOG.md]: This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (104/100 (104%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:60933 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1aflgI-001eIP-1j
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 05:54:26 -0400
Date: Mon, 14 Mar 2016 09:16:28 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1457972188;
	bh=mTQsuSqeyI1ADA7y9OgotveHtTESE7NKVyHI2YJJ5ac=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=gb9Xf9QXAgDBBrksVgL1wCBXc9Nsw5zVrhen7CCY/Nlccn6PlRBuI862HLdq6D2oi
	 9nTw0XAAuBe8t/505vQKkivRlNhFBb34OMxvZk+hGjdr6ZmD9IutM+51xNBeZ/pGPX
	 y2HoDWnRgges0uCAGky0IPaCUbyuOKCCnfUPGCXY=

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java (hunk 3 above)
Comment (jcj52436999) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java]: This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (108/100 (108%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext6.iad.github.net ([192.30.252.197]:32947 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1aflj5-001fci-8j
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 05:57:19 -0400
Date: Mon, 14 Mar 2016 09:18:10 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1457972290;
	bh=SGUnlyhVXk3OeOSzwDLxb6wn5xzhSOB/C7rSAK9Oby8=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=eCfx9kIA7cwcaYMcrZFZgxkb0fC3gkWLf7mStqWq9u2S1zGcH7mYeLsuuXgKUtUQx
	 bc+hbeog2HYuYzCEFSF7mXsG7aA8QlqT6DBNDAegrFEj0sbp3VSbPtM9M4sV8Xpweq
	 c/uvbojYoh69UixAXfuV1/3d2wSqiIpI5HORGjeg=

**Comment by jcj52436999** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (110/100 (110%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext1.iad.github.net ([192.30.252.192]:53768 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1aflj9-001fe5-8n
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 05:57:23 -0400
Date: Mon, 14 Mar 2016 09:21:28 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1457972488;
	bh=/hfJ0bnEgF1GHJ22ZrwEwAwy69VJU2PmZCSnEhs6ssI=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=YWe6znI+qjiyUSA6h/WNo9O6rabn6kmiJ4ABKUacjAWxoTRP4cD5iTB/SGJggmc4T
	 JofM2NjZ8nFjE7/6lUOo4BjPtZ7E7nepM5HmAETHIy/JL95U0HESqME6IvKFK4vWvJ
	 b06nxSQOrztb62iBwdVDFDGxa0c63gDClSV6FMr4=

**Comment by jcj52436999** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (112/100 (112%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext2.iad.github.net ([192.30.252.193]:36336 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1aflji-001fuY-TC
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 05:57:58 -0400
Date: Tue, 15 Mar 2016 00:30:31 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1458027031;
	bh=kNt2XlzxOTTWgzL9JuGQEEzuA51GNUvzq+f4N1QX94I=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=mJhGkLoFZXDt/AF1qvkmje1MzoiVjSr0YhKd0LlKC8JT+NtaGIruCRX6KJrhok8Cn
	 DC/bKwQlJB4vW4f5VjGgZmfPqrfS2AU9yzpoADLCSWdNe9W5ooNFu+p2gmGR4c/HD7
	 ummZ+5VEle1LTKhhdDHVBtzHe1qEgPjUiPORAYNA=

**Comment by jcj52436999** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (119/100 (119%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:38687 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1aflk9-001g99-Ot
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 05:58:25 -0400
Date: Tue, 15 Mar 2016 00:30:38 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1458027038;
	bh=hM9b/6lmaUcYc5IRwlvBMhF0ID+zMO8OlFmLKdraNpY=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=Kyt2XeLosgHb89EBHrdzco8d+y581g2UzHtoLv8yCmhpMUgYwTqCc22zhteW6R2GB
	 csil4hUFvTnhuKBg+jRW906U7JFY5tOQc4qFfFHcKt3aLiFTg3+qWikn3fRZ7m1HbB
	 K91925cJvoaKQ0W/8TYXn9mXt7YLov5bcPrtalVQ=

**Comment by jcj52436999** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (124/100 (124%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:34497 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1aflkk-001gQr-UL
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 05:59:02 -0400
Date: Tue, 15 Mar 2016 00:30:33 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1458027033;
	bh=nSC2ovXTE7NNmaBra8CIpSdNhT46aKTajxInTqCofQ4=;
	h=From:Reply-To:To:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=HrahFwAM/qVFeUtMA2GjOEujd1pKmIqJYAf5ppbuSXu/mqjdKUqA+nLAoKhUuQg6i
	 5Eo8nvugIdY6aAKrJqC0ovPqj6xHd+9nPyo95UVxo9XXQHvd95/AvOCj3KBopN9O/b
	 ZCe3tmjMPJffA2+fW5XxmR1mSSheTcUrYOLMV06Y=

**Comment by jcj52436999** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (123/100 (123%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:41967 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1afmO2-001y8Z-0w
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 06:39:38 -0400
Date: Mon, 14 Mar 2016 03:01:17 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1457949678;
	bh=DxsIFx9egYtmGIdP9MTio6ZgEDL9NxqNTbHZGGB5ric=;
	h=From:Reply-To:To:Subject:List-ID:List-Archive:List-Post:
	 List-Unsubscribe:From;
	b=b2t3cvOI15JkrdDAvR0GGajrDTrzesrX3Xd/nqZhSKjtbUNxa96pRzPh1yvc8tryt
	 b8Z3hDp3EoPNuTCXWxS2jZppV6Jxj8ZKsPCqAL4pdHJENF9uDBnVwrJAYIv046rS0q
	 wt+13ouzZwSS/0eCQ2s0/gPo+7f+EzpKpffDW4WM=

**Pull request merged**